### PR TITLE
add support for Bech32-encoded, TLV-encoded identifiers (nprofile, nevent, nrelay, naddr)

### PR DESCRIPTION
**Pull request by bryanmontz**
See NIP-19:
https://github.com/nostr-protocol/nips/blob/master/19.md

Closes https://github.com/nostr-sdk/nostr-sdk-ios/issues/112

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ disabled_rules:
   - nesting
   - trailing_whitespace
   - type_body_length
+  - function_body_length
```
Comment (tyiu) [inline .swiftlint.yml:9]: Nit: sort alphabetically.

File: Sources/NostrSDK/Bech32.swift [hunk 2]
```diff
@@ -106,9 +106,10 @@ class Bech32 {
         guard let strBytes = str.data(using: .utf8) else {
             throw DecodingError.nonUTF8String
         }
-        guard strBytes.count <= 90 else {
-            throw DecodingError.stringLengthExceeded
-        }
+        // Montz: The length limit of 90 characters is for bitcoin and must be removed for Nostr identifiers, which can be longer.
```
Comment (tyiu) [inline Sources/NostrSDK/Bech32.swift:109]: Nice comment.

File: Tests/NostrSDKTests/Bech32Tests.swift [hunk 3]
```diff
@@ -31,7 +31,8 @@ class Bech32Tests: XCTestCase {
     private let _invalidChecksum: [InvalidChecksum] = [
         (" 1nwldj5", Bech32.DecodingError.nonPrintableCharacter),
         ("\u{7f}1axkwrx", Bech32.DecodingError.nonPrintableCharacter),
-        ("an84characterslonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1569pvx", Bech32.DecodingError.stringLengthExceeded),
+        // Montz: The length limit of 90 characters is for bitcoin and must be removed for Nostr identifiers, which can be longer.
+//        ("an84characterslonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1569pvx", Bech32.DecodingError.stringLengthExceeded),
```
Comment (tyiu) [inline Tests/NostrSDKTests/Bech32Tests.swift:35]: I know we copied this test but should we add this test case to `_validChecksum` with the same explanation comment?

**Review by tyiu — COMMENTED** [review]
Reviewed half of the PR, but had to stop because I just boarded a flight. Will continue later.

**Comment by bryanmontz** [comment]
Thanks for review, @tyiu! I added an explanation for the decoding algorithm that I think should improve its grok-ability. ;)

File: Sources/NostrSDK/MetadataCoding.swift [hunk 4]
```diff
@@ -83,6 +89,33 @@ public extension MetadataCoding {
     /// - Returns: The metadata decoded from the identifier.
     ///
     /// > Note: This function is provided for debugging and testing, as it is an intermediate result.
+    ///
+    /// Given the example profile identifier from NIP-19:
```
Comment (tyiu) [inline Sources/NostrSDK/MetadataCoding.swift:93]: Really nice documentation! Thanks!

**Review by tyiu — APPROVED** [review]
Great

**Pull request merged**